s2i/builder.go

```diff
@@ -372,7 +372,7 @@ func builderImage(f fn.Function) (string, error) {
 		return "", ErrRuntimeRequired
 	}
 
-	v, ok := f.BuilderImages[fn.BuilderS2i]
+	v, ok := f.BuilderImages["s2i"]
```

Comment:
I really prefer one place with a constants for this kind of fields, than some random strings spread across the codebase.

Reply:
Sometimes strings are better than constants.  I won't try to defend them in this case, and am happy to put them back in.  One thing worth ponderig: the core package has only interfaces, and knows nothing of the various implementations of the Builder interface which exist elsewhere.  Placing a constant there may cause confusion in the future if we are not diligent to keep the core clean of dependencies :)